### PR TITLE
Add business-status and authored-on params to query API.

### PR DESCRIPTION
**Pull request by mfjarvis**

### CODE REVIEW

File: coordinator/src/routes/tracker/task.ts [hunk 1]
```diff
@@ -8,34 +8,66 @@ import {getScope, RequestHeaders} from "../../utils/headers"
...
+  "business-status"?: string
+  "authored-on"?: string | Array<string>
 }
 
-export type ValidQuery = Partial<Record<QueryParam, string>>
-
-interface QueryParamProperties {
+export interface QueryParamDefinition {
+  name: keyof ValidQuery,
+  supportedBySpine: boolean
```
Comment (mfjarvis) [inline coordinator/src/routes/tracker/task.ts:24]: Needs a better name. I want something which means "Spine allows queries using this as the only parameter", but the current name suggests something more like "Spine understands this parameter".

File: coordinator/src/routes/tracker/task.ts [hunk 2]
```diff
@@ -74,42 +106,86 @@ export default [{
...
-  const pipeIndex = rawValue?.indexOf("|") || -1
-  if (pipeIndex !== -1) {
-    return rawValue.substring(pipeIndex + 1)
-  }
-  return rawValue
+  throw new Error("Attempting to make tracker request without prescription or patient identifier")
 }
 
 export function filterBundleEntries(bundle: fhir.Bundle, queryParams: ValidQuery): void {
+  console.log(`Got ${bundle.total} results from Spine before filtering.`)
```
Comment (mfjarvis) [inline coordinator/src/routes/tracker/task.ts:134]: Remove or use request.logger

File: coordinator/src/services/validation/query-validator.ts [hunk 3]
```diff
@@ -12,43 +15,68 @@ export const validateQueryParameters = (
...
+  return issues
+}
+
+function validateQueryParameter(
+  definition: QueryParamDefinition,
+  values: Array<string>
+): Array<fhir.OperationOutcomeIssue> {
+  const issues: Array<fhir.OperationOutcomeIssue> = []
+
+  if (!definition.isDateParameter && values.length > 1) {
```
Comment (mfjarvis) [inline coordinator/src/services/validation/query-validator.ts:48]: Not sure if this is actually an error according to the FHIR spec

**Comment by mfjarvis** [comment]
EPSAT changes will be made in another PR because EPSAT PR deploys point at the master branch deploy.

File: coordinator/src/routes/tracker/task.ts (hunk 1 above)
Comment (dan-jones-iw) [reply coordinator/src/routes/tracker/task.ts:24]: `validInIsolation`?

File: coordinator/src/services/validation/query-validator.ts (hunk 3 above)
Comment (mfjarvis) [reply coordinator/src/services/validation/query-validator.ts]: Will leave for now

**Review by dan-jones-iw — APPROVED** [review]

**Pull request merged**